revpi-sos

```diff
@@ -40,22 +55,25 @@ Are you going to continue (N/y)?" -n 1 t_ant
 fi
 
 echo "report generating..."
-SOSFILE=$(sudo sos report -o revpi,revpi-eep,revpi-codesys --batch | grep "/tmp/sosreport" | sed -e 's/^[ \t]*//')
-echo
-echo "Your report has been generated and saved in:"
-echo "$SOSFILE"
-echo
+sudo sos report -o revpi,revpi-eep,revpi-codesys --batch --tmp-dir "$sos_tmp_dir" >/dev/null
+sos_file=$(ls "$sos_tmp_dir"/*.tar.xz)
```

Comment:
Why are you not referencing the correct file? Why the *? And why the `ls`? This will break the other code, if there is more then one file that matches.

Reply:
It should also be noted that the output of `ls` should never be used in scripts in the first place.

Reply:
The actual filename is generated by **sos report** and is not known beforehand. There is also no clean to get the filename from the command as it just prints it within other text in its verbose output.

Thats why each report is now created in a unique subfolder. In this folder there can be only one **.tar.xz** file which is identified using **ls** on that subfolder instead of parsing the quite verbose output of **sos report**

Reply:
As I said before, I can spray the /tmp/ with revpi-sos-.... directories. Then you call revpi-sos-report and the script just keeps using my directories!

Reply:
> In this folder there can be only one .tar.xz file

That's not correct - a different user could write a file with the `.tar.xz` suffix into that directory. We're (possibly, likely even) in `/tmp` here, anyone can put files there.